include/mbedtls/threading.h

```diff
@@ -39,16 +39,72 @@ extern "C" {
 #define MBEDTLS_ERR_THREADING_BAD_INPUT_DATA              -0x001C  /**< Bad input parameters to function. */
 #define MBEDTLS_ERR_THREADING_MUTEX_ERROR                 -0x001E  /**< Locking / unlocking / free failed with error code. */
 
+/* Forwards declaration of mutex type  */
+typedef struct _mbedtls_threading_mutex_t mbedtls_threading_mutex_t;
```

Comment:
This forward declaration actually causes a redefinition problem with line https://github.com/ARMmbed/mbedtls/pull/697/files#diff-56eb87e4b150ad36491122269dae6eb1R105, and probably the same would happen with line https://github.com/ARMmbed/mbedtls/pull/697/files#diff-56eb87e4b150ad36491122269dae6eb1R144. This is partly why the Travis CI is failing.